aten/src/ATen/Context.cpp

```diff
@@ -64,6 +64,23 @@ bool Context::deterministic() const {
 }
 
 void Context::setDeterministic(bool b) {
+  if (b && hasCUDART() && (versionCUDART() >= 10020)) {
+    // If using CUDA 10.2 or greater, need to make sure CuBLAS workspace config
+    // is set to deterministic setting
+    static const char config_var_name[] = "CUBLAS_WORKSPACE_CONFIG";
+    static const char config0[] = ":4096:8";
+    static const char config1[] = ":16:8";
+    char* workspace_config = std::getenv(config_var_name);
+    bool workspace_config_is_deterministic = (workspace_config != nullptr) && (
+      (strcmp(workspace_config, config0) == 0) || (strcmp(workspace_config, config1) == 0)
+    );
+    TORCH_CHECK(workspace_config_is_deterministic,
+      "To enable deterministic behavior with CUDA >= 10.2, you must set environment variable ",
+      config_var_name, "=", config0, " or ",
+      config_var_name, "=", config1, ". For more information, go to ",
+      "https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility"
+    );
```

Comment:
This seems questionable to me. If I never call any code that exercises cuBLAS I shouldn't get an error like this, right? But yet, I get a HARD error in this case. Maybe this error test should be more lazy?

Reply:
Yeah that's a good point, I'll make that change. Looks like this PR is going to get a lot bigger